llama_index/embeddings/gradient.py

```diff
@@ -0,0 +1,95 @@
+# Reference: https://huggingface.co/BAAI/bge-large-en-v1.5#model-list
+QUERY_INSTRUCTION_FOR_RETRIEVAL = (
+    "Represent this sentence for searching relevant passages:"
+)
+
+
+class GradientEmbedding(BaseEmbedding):
+    """GradientAI embedding models.
+
+    This class provides an interface to generate embeddings using a model
+    deployed in Gradient AI. At the initialization it requires a model_id
+    of the model deployed in the cluster.
+
+    Note:
+        Requires `gradientai` package to be available in the PYTHONPATH. It can be installed with
+        `pip install gradientai`.
+    """
+
+    _gradient: Any = PrivateAttr()
+    _model: Any = PrivateAttr()
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "GradientEmbedding"
+
+    def __init__(
+        self,
+        gradient_access_token: str,
+        gradient_workspace_id: str,
+        gradient_model_slug: str,
+        gradient_host: Optional[str] = None,
+        **kwargs: Any,
+    ):
+        """Initializes the GradientEmbedding class.
+
+        During the initialization the `gradientai` package is imported. Using the access token,
+        workspace id and the slug of the model, the model is fetched from Gradient AI and prepared to use.
+
+        Args:
+            gradient_access_token (str): The access token of the Gradient AI account.
+            gradient_workspace_id (str): The workspace ID of the Gradient AI account.
+            gradient_model_slug (str): The model slug of the model in the Gradient AI account.
+            gradient_host (str, optional): The host of the Gradient AI API. Defaults to None, which
+              means the default host is used.
+
+        Raises:
+            ImportError: If the `gradientai` package is not available in the PYTHONPATH.
+            ValueError: If the model cannot be fetched from Gradient AI.
+        """
+        try:
+            import gradientai
+        except ImportError:
+            raise ImportError("GradientEmbedding requires `pip install gradientai`.")
+
+        self._gradient = gradientai.Gradient(
+            access_token=gradient_access_token,
+            workspace_id=gradient_workspace_id,
+            host=gradient_host,
+        )
+
+        try:
+            self._model = self._gradient.get_embeddings_model(slug=gradient_model_slug)
+        except gradientai.openapi.client.exceptions.UnauthorizedException as e:
+            logger.error(f"Error while loading model {gradient_model_slug}.")
+            self._gradient.close()
+            raise ValueError("Unable to fetch the requested embeddings model") from e
+
+        super().__init__(model_name=gradient_model_slug, **kwargs)
+
+    async def _aget_query_embedding(self, query: str) -> Embedding:
+        # Gradient AI doesn't have the proper API for async yet, so we just use the sync version.
+        return self._get_query_embedding(query)
```

Comment:
would rather just give a not yet implemented error or something

Reply:
I was looking at other implementations and actually most of them implement that way :) I guess we should add async soon, tbh.